src/main/kotlin/sliv/tool/scene/SceneFacade.kt

```diff
@@ -0,0 +1,78 @@
+package sliv.tool.scene
+
+import javafx.scene.image.Image
+import sliv.tool.project.model.*
+import sliv.tool.scene.controller.SceneController
+import sliv.tool.scene.model.*
+import java.io.FileInputStream
+
+class SceneFacade(private val controller: SceneController) {
+    private val visualizationLayers = HashMap<String, Layer>()
+
+    fun visualize(layers: List<ProjectLayer>, frames: List<ProjectFrame>) {
+        layers.forEach { projectLayer ->
+            val visualizationLayer = visualizationLayers[projectLayer.name]
+            if (visualizationLayer != null && visualizationLayer.kind == projectLayer.kind) {
+                return //don't rewrite existing settings
+            }
+            visualizationLayers[projectLayer.name] = projectLayer.toVisualizationLayer()
+        }
+        val visualizationLayers = layers.map { it.toVisualizationLayer() }
+        val scene =
+            Scene({ i -> frames[i].toVisualizationFrame() }, frames.count(), visualizationLayers)
+        controller.scene.value = scene
+    }
+
+    private fun ProjectLayer.toVisualizationLayer(): Layer {
+        return when (kind) {
+            LayerKind.KEYPOINT -> Layer.PointLayer(this.name)
+            LayerKind.LINE -> Layer.LineLayer(this.name)
+            LayerKind.PLANE -> Layer.PlaneLayer(this.name)
+        }
+    }
+
+    private fun ProjectFrame.toVisualizationFrame(): VisualizationFrame {
+        val image = Image(FileInputStream(this.imagePath.toFile()))
+        val landmarks = HashMap<Layer, List<Landmark>>()
+        landmarkFiles.forEach { file ->
+            landmarks[visualizationLayers[file.projectLayer.name]!!] =
```

Comment:
remove `!!` -- better check it before and skip file with log if nothing found that crash